### PR TITLE
test: fix portfolio spec by introducing toPass with intervals

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Buttons not being clickable might have something to do with a hydration issue. Mainly when the button UI is loaded but the functionality of the button itself is not. Making the button not clickable.

Here's a video that explains it:
https://www.youtube.com/watch?v=8g7FvoRToGo

### CODE REVIEW

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

File: e2e/portfolio.spec.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ test.afterAll(() => {
 
 test.describe('Add Portfolio Item', () => {
   test.beforeEach(async ({ page }) => {
+    test.setTimeout(61000);
```
Comment (ojeytonwilliams) [inline e2e/portfolio.spec.ts:17]:
```suggestion
```
Is this necessary? It's a very long time for a test.

File: e2e/portfolio.spec.ts [hunk 2]
```diff
@@ -27,6 +28,13 @@ test.describe('Add Portfolio Item', () => {
     await page
       .getByRole('button', { name: 'Add a new portfolio Item' })
       .click();
+
+    // Will check if the portfolio button is hydrated correctly with different intervals.
+    await expect(async () => {
+      await expect(
+        page.getByRole('button', { name: 'Add a new portfolio Item' })
+      ).toBeDisabled();
+    }).toPass();
```
Comment (ojeytonwilliams) [inline e2e/portfolio.spec.ts:37]:
```suggestion

    // Will check if the portfolio button is hydrated correctly with different intervals.
    await expect(async () => {
      const addPortfolioItemButton = page.getByRole('button', {
        name: 'Add a new portfolio Item'
      });
      await addPortfolioItemButton.click();

      await expect(addPortfolioItemButton).toBeDisabled({ timeout: 1 });
    }).toPass();
```
So, the thing is, there's no point re-checking if the button is disabled without clicking it. The problem is that if playwright clicks too early, the button click doesn't do anything, so waiting for even longer (i.e. what `toPass` does) just makes the test take longer to fail.

The way to think about `toPass` is that it lets you retry something unreliable (clicking) until it finally succeeds (button is disabled). You have to do both in the callback for that to work.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

**Comment by Sembauke** [comment]
I had tried this yesterday but it didn’t work. The timeout change is needed because the interval of toPass is longer.

File: e2e/portfolio.spec.ts (hunk 2 above)
Comment (Sembauke) [reply e2e/portfolio.spec.ts]: I must not have been thinking straight 🙃

**Comment by Sembauke** [comment]
One more attempt 🙄

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Review by ShaunSHamilton — APPROVED** [review]
👍 Thank you for the link; I learnt something new.

**Pull request merged**